python/lsst/pipe/analysis/visitAnalysis.py

```diff
@@ -550,22 +550,35 @@ def readCatalogs(self, dataRefList, dataset, repoInfo, aliasDictList=None, fakeC
 
             # Compute Focal Plane coordinates for each source if not already there
             if self.config.hasFakes:
-                exp = repoInfo.butler.get("calexp", dataRef.dataId)
+                # Get just a single pixel of the exposure
+                # Exposure is needed for ExposureInfo including bounding
+                # polygon
+                pix = lsstgeom.Box2I(lsstGeom.Point2I(0, 0),
+                                     lsstGeom.Point2I(0, 0))
+                exp = butler.get("calexp_sub", dataId, bbox=pix)
                 wcs = exp.getWcs()
-                mask = exp.mask.array
-                maskBad = mask & 2**0
-                good = len(np.where((maskBad == 0))[0])
-                pixScale = exp.getWcs().getPixelScale().asArcseconds()
+
+                # Hacky but fast way of getting just the mask
+                # Actual mask is needed because BAD pixels are more than
+                # just the defects
+                fname = butler.get("calexp_filename", dataId)[0]
+                mask = afwImage.Mask(fname, hdu=2)
```

Comment:
If we didn't want to specify an hdu explicitly would `ExposureFitsReader.readMask` work better?

Reply:
Yes.  And I hadn't noticed that there's an `ExposureFitsReader.readExposureInfo` that would help, too.